src/cards/merge.ts

```diff
@@ -0,0 +1,194 @@
+import { makeManaTrail } from '../graphics/Particles';
+
+const merge_id = 'merge';
+const spell: Spell = {
+  card: {
+    id: merge_id,
+    category: CardCategory.Soul,
+    manaCost: 60,
+    healthCost: 0,
+    probability: probabilityMap[CardRarity.RARE],
+    expenseScaling: 1,
+    thumbnail: 'spellIconMerge.png',
+    description: 'spell_merge',
+    effect: async (state, card, quantity, underworld, prediction) => {
+      // Batch find targets that should be merged
+      // Note: They need to be batched because the list will be changed as units are merged
+      const targets = getCurrentTargets(state);
+      let mergedTargets: HasSpace[] = [];
+
+      // We have to loop by target instead of unit type here,
+      // because there is other criteria to consider:
+      // we don't want to merge living units into dead ones, for example
+      for (let i = 0; i < targets.length; i++) {
+        const target = targets[i];
+        if (!target || mergedTargets.includes(target)) continue;
+
+        // if target has not been merged already
+        // find similar and merge those into target
+        const potentialTargets = targets.filter(t => !mergedTargets.includes(t));
+        const similarThings = findSimilar(target, underworld, prediction, potentialTargets);
+
+        if (similarThings.length) {
+          if (!prediction) {
+            playSFXKey('clone');
+            for (const thing of similarThings) {
+              makeManaTrail(thing, target, underworld,
+                colors.convertToHashColor(colors.manaBrightBlue),
+                colors.convertToHashColor(colors.manaDarkBlue),
+                1);
+            }
+            await animateMerge((target as any).image);
+          }
+
+          mergedTargets = mergedTargets.concat(similarThings);
+          if (Unit.isUnit(target)) {
+            const similarUnits = similarThings as Unit.IUnit[];
+            mergeUnit(target, similarUnits, underworld, prediction);
+          } else if (Pickup.isPickup(target)) {
+            const similarPickups = similarThings as Pickup.IPickup[];
+            mergePickup(target, similarPickups, underworld, prediction);
+            mergedTargets.concat(similarPickups);
+          }
+        }
+      }
+
+      if (!mergedTargets.length) {
+        refundLastSpell(state, prediction, 'Target things of the same type!')
+      }
+      return state;
+    },
+  },
+};
+
+export function mergeUnit(target: Unit.IUnit, unitsToMerge: Unit.IUnit[], underworld: Underworld, prediction: boolean) {
+  // TODO - I forsee a bug that causes modifier addition to change game state before merge is complete
+  // I.E. Target gets a few stacks of suffocate and dies, bloat triggers, bunch of stuff changes.
+  // Loop keeps going and reaches some undefined value, or tries to add modifiers to dead target, etc.
+  for (const unit of unitsToMerge) {
+    // Never merge a player controlled unit
+    if (unit.unitType == UnitType.PLAYER_CONTROLLED) continue;
```

Comment:
I feel like _something_ should happen if you try to merge players.  Better to do something cool even if it ends the run than do nothing (fizzle spells are not fun in a game about spell combining).  If you try to merge with Non player spellmasons (like clones) maybe it just adds to your current health/mana/stam rather than max?

If you try to merge with another player... I'm not sure, we can't destroy other players units but I know people are going to try it so it should do something.  Maybe it just steals their current health and mana down to 1 and gives them to the caster?  We should definitely do something to pleasantly surprise players who try it

Reply:
@jdoleary 
Okay, hear me out on this one: When a player would be merged into another unit, it instead spawns a little tiny Spellmason 🤣

Reply:
Hahaha I love it! Like between them? or how would it work?

Reply:
The simplest version I can think of is spawning a weaker/smaller Spellmason near the main or "merged into" target (Imagine a spellmason with split or split 2 on it, for example). Spawning it in the middle of the two players could cause some weird situations with terrain and whatnot.

Reply:
Hmm, I do really like that idea from a gameplay standpoint, and it seems a little more intuitive / in-line with the regular spell behavior, but I wonder if it's worth giving up the tiny Spellmason? After showing a couple people, it feels like one of those "oh my gosh!" laughing moments between friends, and a feature people would adore and talk about, so I'm wondering if that's more valuable here? What do you think? Maybe we disable merging between ai/player units, and keep this in as a player-player easter egg?

Reply:
I like the idea of the tiny spellmason but it doesn't make sense to me functionally.  Like I think it'll feel like a bug.  I wonder if there's another way we can introduce that.  Merge doesn't create any new units under any other circumstances.
Let's go with my last take: " it kills the other player (leaving their body) and gives their stats (current stats, not max stats) to the caster. "
I think this makes the most sense given the description of the spell.  And we can backburner the tiny spellmason idea, maybe we can make it an easter egg some other way

Reply:
Done. One visual issue: Predictions don't show the stamina a player would gain by merging.